scripts/benchmark_all.sh

```diff
@@ -0,0 +1,152 @@
+end_commit_reference=$2
+shift; shift
+start_commit=$(git rev-parse $start_commit_reference | head -n 1)
+end_commit=$(git rev-parse $end_commit_reference | head -n 1)
+
+# Check status of repository
+if [[ $(git status --untracked-files=no --porcelain) ]]
+then
+  echo 'Cowardly refusing to execute on a dirty workspace'
+  exit 1
+fi
+
+# Ensure that this is run on a release build. Not really necessary, just a sanity check.
+output=$(grep 'CMAKE_BUILD_TYPE:STRING=Release' CMakeCache.txt || true)
+if [ -z "$output" ]
+then
+  echo 'Current folder is not configured as a release build'
+  exit 1
+fi
+
+# Check whether to use ninja or make
+output=$(grep 'CMAKE_MAKE_PROGRAM' CMakeCache.txt | grep ninja || true)
+if [ ! -z "$output" ]
+then
+  build='ninja'
+else
+  build='make'
+fi
+
+# Create the result folder if necessary
+mkdir benchmark_all_results 2>/dev/null || true
+
+build_folder=$(pwd)
+
+# Here comes the actual work
+for commit in $start_commit $end_commit
+do
+  # Skip commits where we already have all results
+  if [ -f "benchmark_all_results/complete_${commit}" ]
+  then
+    echo "Benchmarks have already been executed for ${commit}. Skipping."
+    echo "Run rm benchmark_all_results/*${commit}* to re-execute."
+    sleep 1  # Easiest way to make sure the above is read
+    continue
+  fi
+
+  # Checkout and build from scratch, tracking the compile time
+  git checkout $commit
+  git submodule update --init
+  echo "Building $commit..."
+  $build clean
+  /usr/bin/time -p sh -c "( $build -j $(nproc) ${benchmarks} 2>&1 ) | tee benchmark_all_results/build_${commit}.log" 2>"benchmark_all_results/build_time_${commit}.txt"
+
+  # Run the benchmarks
+  cd ..  # hyriseBenchmarkJoinOrder needs to run from project root 
+  for benchmark in $benchmarks
+  do
+    echo "Running $benchmark for $commit... (single-threaded)"
+    ( ${build_folder}/$benchmark -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.log"
+
+    echo "Running $benchmark for $commit... (multi-threaded)"
+    ( ${build_folder}/$benchmark --scheduler --clients ${num_mt_clients} -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt.log"
+  done
+  cd "${build_folder}"
+
+  # After all benchmarks are done, leave a marker so that this commit can be skipped next time
+  touch "benchmark_all_results/complete_${commit}"
+done
+
+# Print the results
+
+echo ""
+echo "==========="
+echo ""
+
+# Print information about the system
+echo "**System**"
+echo "<details>"
+echo "<summary>$(hostname) - click to expand</summary>"
+echo ""
+echo "| property | value |"
+echo "| -- | -- |"
+echo "| Hostname | $(hostname) |"
+if [[ "$(uname)" == 'Darwin' ]]; then
+  echo "| Model | $(defaults read ~/Library/Preferences/com.apple.SystemProfiler.plist 'CPU Names' | cut -sd '"' -f 4) |"
```

Comment:
```bash
$ defaults read ~/Library/Preferences/com.apple.SystemProfiler.plist 'CPU Names'
{
    "JHD4-en-DE_DE" = "MacBook Pro (13-inch, 2018, Four Thunderbolt 3 Ports)";
    "ML7M-en-DE_DE" = "MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports)";
}
```
Special case since I have upgraded my system previously. Consider using the following:
```bash
$ sysctl -n hw.model
MacBookPro16,2
```

Reply:
Interesting. I took the code from StackExchange, where people seem to be unaware of this. Added `tail -n1` and added a comment over there.

```
$ sysctl -n hw.model
MacBookPro16,2
```

Looked at it before, hated it. What does 16 mean? Apparently neither the year, nor the display size.

Reply:
> Looked at it before, hated it. What does 16 mean? Apparently neither the year, nor the display size.

They seem to count up with every new generation. Here is a [list](https://everymac.com/systems/by_capability/mac-specs-by-machine-model-machine-id.html).  But it is not nice. So stay with the above and `tail -n1` ;)